src/circuit.rs

```diff
@@ -397,12 +399,18 @@ impl plonk::Circuit<pallas::Base> for Circuit {
 
         // Merkle path validity check.
         let anchor = {
+            let path = self.path.map(|typed_path| {
+                // TODO: Replace with array::map once MSRV is 1.55.0.
+                gen_const_array_with_default(MerkleHashOrchard::empty_leaf().inner(), |i| {
+                    typed_path[i].inner()
+                })
```

Comment:
Here we _could_ use the default version, as `pallas::Base` implements `Default`:
```suggestion
                gen_const_array(|i| typed_path[i].inner())
```